File: nekoshlink/shlink-security/src/main/java/org/nekosoft/shlink/sec/audit/AccessLogger.java
package org.nekosoft.shlink.sec.audit;

import kotlin.Unit;
import mu.KLogger;
import mu.KotlinLogging;
import org.springframework.context.event.EventListener;
import org.springframework.security.authentication.event.AbstractAuthenticationFailureEvent;
import org.springframework.security.authentication.event.AuthenticationSuccessEvent;
import org.springframework.stereotype.Component;


@Component
class AccessLogger {

    private static KLogger kLogger = KotlinLogging.INSTANCE.logger(() -> Unit.INSTANCE);

    @EventListener
    public void onSuccess(AuthenticationSuccessEvent event) {
        kLogger.warn("Successful access attempt for {}", event.getAuthentication());
    }

    @EventListener
    public void onFailure(AbstractAuthenticationFailureEvent event) {
        kLogger.warn("Failed access attempt for {} due to {}", event.getAuthentication(), event.getException().getMessage());
    }

}

File: nekoshlink/shlink-security/src/main/kotlin/org/nekosoft/shlink/sec/audit/AccessLogger.kt
package org.nekosoft.shlink.sec.audit

import mu.KotlinLogging
import org.springframework.boot.actuate.audit.listener.AuditApplicationEvent
import org.springframework.context.event.EventListener
import org.springframework.security.authentication.event.AbstractAuthenticationFailureEvent
import org.springframework.security.authentication.event.AuthenticationSuccessEvent
import org.springframework.stereotype.Component

private val kLogger = KotlinLogging.logger {}

@Component
class AccessLogger {

    @EventListener
    fun onSuccess(event: AuthenticationSuccessEvent) {
        kLogger.warn("Successful access attempt for {}", event.authentication)
    }

    @EventListener
    fun onFailure(event: AbstractAuthenticationFailureEvent) {
        kLogger.warn("Failed access attempt for {} due to {}", event.authentication, event.exception.message)
    }

    @EventListener
    fun on(event: AuditApplicationEvent) {
        kLogger.warn("An Audit Event was received: {}", event)
    }

}